hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -128,6 +129,11 @@ private void validateRequiredFields(Configuration conf, TableSchema schema) {
       throw new ValidationException("Field " + preCombineField + " does not exist in the table schema."
           + "Please check 'write.precombine.field' option.");
     }
+
+    if (conf.getString(FlinkOptions.TABLE_TYPE).toUpperCase().equals(FlinkOptions.TABLE_TYPE_MERGE_ON_READ)
+        && conf.getBoolean(FlinkOptions.INSERT_ALLOW_DUP)) {
+      throw new ValidationException("Option 'write.insert.allow_dup' is only allowed for COPY_ON_WRITE table.");
```

Comment:
Instead of throw exception, how about we just change the option `write.insert.allow_dup` to false and log warnings there.

Reply:
This may be lead users to misunderstand, I think we shouldn't modify the parameters.